### PR TITLE
Make dependencies in Bochs and x3270 optional

### PR DESCRIPTION
**Pull request by ronmrdechai**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/bochs.rb [hunk 1]
```diff
@@ -48,6 +48,10 @@ def install
       --with-term
     ]
 
+    if build.with? "sdl2"
```
Comment (DomT4) [inline Formula/bochs.rb]: Don't need the block if it's just one argument, FWIW.
```ruby
args << "--with-sdl2" if build.with? "sdl2"
```

File: Formula/x3270.rb [hunk 2]
```diff
@@ -10,17 +10,18 @@ class X3270 < Formula
...
   option "with-pr3287", "Include pr3287 (printer emulation)"
 
-  depends_on :x11
+  depends_on :x11 => :recommended
   depends_on "openssl"
 
   def install
     args = ["--prefix=#{prefix}"]
-    args << "--enable-x3270"
+    args << "--enable-x3270" if build.with? :x11
```
Comment (DomT4) [inline Formula/x3270.rb]: `"x11"` not `:x11` here.

**Comment by DomT4** [comment]
You'll need to squash down to one commit per formula & rebase here, CI is having some issues processing this PR 😢.

**Comment by ronmrdechai** [comment]
Not really sure how I can do that, I'll try. Thanks!

**Pull request closed without merging**